CredentialProvider.Microsoft/CredentialProviders/KeyVault/KeyVaultHelper.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+//
+// Licensed under the MIT license.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Resources = NuGetCredentialProvider.Resources;
+
+namespace Microsoft.Azure.KeyVault.Helper
+{
+    /// <summary>
+    /// This class provides access to secrets stored in the KeyVault
+    /// </summary>
+    public sealed class KeyVaultHelper : IDisposable
+    {
+        public struct Config
+        {
+            public string CertificateThumbprint { get; set; }
+            public string CertificateStoreType { get; set; }
+            public string KeyVaultUrl { get; set; }
+            public string ClientId { get; set; }
+            public bool? UseMsi { get; set; }
```

Comment:
Is this a customer facing config? I would consider expanding this acronym considering it's not all that common.

Reply:
This is not an external data structure unless one wants to reuse KeyVaultHelper class with a different project. I can make it internal. Customer-facing config is in ConfigManager, string keys for key-value pairs in the config. 
I use the following values:
KeyVaultUrl
KeyVaultUseMsi
KeyVaultAuthClientId
KeyVaultAuthCertificateStoreType
KeyVaultAuthCertificateThumbprint

I can make KeyVaultUseMsi implicit when CertificateThumbprint is not specified, or can expand it to KeyVaultUseManagedServerIdentity.